Python/optimizer.c

```diff
@@ -888,6 +894,10 @@ translate_bytecode_to_trace(
         // Add cache size for opcode
         instr += _PyOpcode_Caches[_PyOpcode_Deopt[opcode]];
 
+        if (opcode == BINARY_OP_INPLACE_ADD_UNICODE) {
```

Comment:
Can you move this up to the `if (uop == _BINARY_OP_INPLACE_ADD_UNICODE)` clause above?
I'd like to keep the special casing code in one place.

You'll probably want to add an assert this uop is the last in the expansion, as well. Like we do in `if (uop == _PUSH_FRAME)` above.